addons/point_of_sale/static/src/js/db.js

```diff
@@ -354,9 +363,12 @@ var PosDB = core.Class.extend({
     get_product_by_barcode: function(barcode){
         if(this.product_by_barcode[barcode]){
             return this.product_by_barcode[barcode];
-        } else {
-            return undefined;
+        } else if (this.product_packaging_by_barcode[barcode]) {
+            var product = this.product_by_id[this.product_packaging_by_barcode[barcode].product_id[0]];
+            product.barcode_packaging_qty = this.product_packaging_by_barcode[barcode].qty;
```

Comment:
There will be problem here if the product is not found. Setting a props to an undefined will crash. I guess you need to do either 1) load the products reference in each packaging after loading the `product.packaging` records; or 2) just return `undefined` here when the product is not found.